### PR TITLE
Kevin/history

### PR DESCRIPTION
**Pull request by kevingreer**

### CODE REVIEW

**Comment by kevingreer** [comment]
#113

File: Podcast/BookmarkViewController.swift [hunk 1]
```diff
@@ -127,6 +127,8 @@ class BookmarkViewController: ViewController, UITableViewDelegate, UITableViewDa
         bookmarksTableViewCell.setPlayButtonToState(isPlaying: true)
         appDelegate.showPlayer(animated: true)
         Player.sharedInstance.playEpisode(episode: episode)
+        let request = CreateListeningHistoryElementEndpointRequest(episodeID: episode.id)
```
Comment (natashaarmbrust) [inline Podcast/BookmarkViewController.swift]: sorry not to be picky but make this createListeningHistoryElementEndpointRequest

File: Podcast/DiscoverTableViewCell.swift [hunk 2]
```diff
@@ -260,6 +260,8 @@ class DiscoverTableViewCell: UITableViewCell {
     func clickToPlayImageButtonPress() {
         guard let episode = episode else { return }
         Player.sharedInstance.playEpisode(episode: episode)
+        let request = CreateListeningHistoryElementEndpointRequest(episodeID: episode.id)
```
Comment (natashaarmbrust) [inline Podcast/DiscoverTableViewCell.swift]: Same here with the rename

File: Podcast/DiscoverViewController.swift [hunk 3]
```diff
@@ -191,6 +191,8 @@ class DiscoverViewController: ViewController, UITableViewDelegate, UITableViewDa
         guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return }
         appDelegate.showPlayer(animated: true)
         Player.sharedInstance.playEpisode(episode: episode)
+        let request = CreateListeningHistoryElementEndpointRequest(episodeID: episode.id)
```
Comment (natashaarmbrust) [inline Podcast/DiscoverViewController.swift]: rename

File: Podcast/EpisodeDetailViewController.swift [hunk 4]
```diff
@@ -111,6 +111,8 @@ class EpisodeDetailViewController: ViewController, EpisodeDetailHeaderViewCellDe
         cell.setPlaying(playing: true)
         appDelegate.showPlayer(animated: true)
         Player.sharedInstance.playEpisode(episode: episode)
+        let request = CreateListeningHistoryElementEndpointRequest(episodeID: episode.id)
```
Comment (natashaarmbrust) [inline Podcast/EpisodeDetailViewController.swift]: rename

File: Podcast/FeedViewController.swift [hunk 5]
```diff
@@ -150,6 +150,8 @@ class FeedViewController: ViewController, UITableViewDelegate, UITableViewDataSo
         cardTableViewCell.setPlayButtonToState(isPlaying: true)
         appDelegate.showPlayer(animated: true)
         Player.sharedInstance.playEpisode(episode: card.episode)
+        let request = CreateListeningHistoryElementEndpointRequest(episodeID: card.episode.id)
```
Comment (natashaarmbrust) [inline Podcast/FeedViewController.swift]: rename

File: Podcast/ListeningHistoryViewController.swift [hunk 6]
```diff
@@ -93,15 +108,46 @@ class ListeningHistoryViewController: ViewController, UITableViewDelegate, UITab
...
-        episodes = Array(repeating: episode, count: 5)
+    func handleRefresh() {
+        fetchEpisodes(refresh: true)
+    }
+    
+    func fetchEpisodes(refresh: Bool) {
+        if refresh {
+            offset = 0
+        }
+        let request = FetchListeningHistoryEndpointRequest(offset: offset, max: pageSize)
```
Comment (natashaarmbrust) [inline Podcast/ListeningHistoryViewController.swift]: rename

File: Podcast/SearchTableViewController.swift [hunk 7]
```diff
@@ -164,6 +164,8 @@ class SearchTableViewController: UITableViewController, SearchEpisodeTableViewCe
         cell.setPlayButtonToState(isPlaying: true)
         appDelegate.showPlayer(animated: true)
         Player.sharedInstance.playEpisode(episode: episode)
+        let request = CreateListeningHistoryElementEndpointRequest(episodeID: episode.id)
```
Comment (natashaarmbrust) [inline Podcast/SearchTableViewController.swift]: rename

**Pull request merged**